modules/remote-state-s3-backend/main.tf

```diff
@@ -0,0 +1,22 @@
+provider "aws" {
+  region = var.region
+}
+
+provider "aws" {
+  alias  = "replica"
+  region = var.replica_region
+}
+
+module "remote_state" {
```

Comment:
yes - the state could be copied, when the backend is changed/configured, by `terraform`

> $ terraform init
> Initializing the backend...
> Acquiring state lock. This may take a few moments...
> Do you want to copy existing state to the new backend?
>   Pre-existing state was found while migrating the previous "local" 
>   backend to the newly configured "s3" backend. No existing state 
>   was found in the newly configured "s3" backend. Do you want to 
>   copy this state to the new "s3" backend? Enter "yes" to copy and 
>   "no" to start with an empty state.
>   Enter a value:

I've learned about that from this article[How to manage Terraform state](https://blog.gruntwork.io/how-to-manage-terraform-state-28f5697e68fa)

Btw. it's much better done for [terragrunt](https://terragrunt.gruntwork.io/) - next time I will give one more chance for this tool (this time I had to limit the learning curve ;) ).